packages/block-editor/src/components/grid/use-grid-layout-sync.js

```diff
@@ -120,20 +146,25 @@ export function useGridLayoutSync( { clientId: gridClientId } ) {
 					},
 				};
 			}
+
+			// Unset the columnStart/rowStart attributes for blocks removed
+			// from the grid.
+			previousBlockOrder?.forEach( ( clientId ) => {
+				if ( ! blockOrder.includes( clientId ) ) {
+					const attributes = getBlockAttributes( clientId );
+					const update = unsetGridPositionAttributes( attributes );
+					if ( update ) {
+						updates[ clientId ] = update;
+					}
+				}
+			} );
 		} else {
 			// When in auto mode, remove all of the columnStart and rowStart values.
 			for ( const clientId of blockOrder ) {
```

Comment:
Something I noticed is that the code always loop through these clientids in auto mode, even if the layout didn't change (one of the other deps might have caused the effect to re-run).

A small optimization might be to also keep track of the previous value of `gridLayout.isManualPlacement` and only remove the attributes when `previousIsManualPlacement` is `true`.

I can make a follow-up if you think that's worth doing.

Reply:
I went ahead and made a PR - [Grid perf: Avoid iterating auto grid inner blocks unless mode specifically changed](https://github.com/WordPress/gutenberg/pull/64194#top)